src/odb/src/db/dbBlock.cpp

```diff
@@ -3351,24 +3351,14 @@ void dbBlock::writeDb(char* filename, int allNode)
       dbname = fmt::format("{}.remote.{}.db", filename, getpid());
   } else
     dbname = fmt::format("{}.db", filename);
-  FILE* file = fopen(dbname.c_str(), "wb");
-  if (!file) {
+  std::ofstream file(dbname, std::ios::binary);
```

Comment:
add missing exception handling to this stream, as the rest of the code assumes it.